lib/stripe/stripe_client.rb

```diff
@@ -229,19 +229,36 @@ def format_app_info(info)
       str
     end
 
-    def handle_api_error(http_resp)
+    def handle_error_response(http_resp)
       begin
         resp = StripeResponse.from_faraday_hash(http_resp)
         error = resp.data[:error]
 
-        unless error && error.is_a?(Hash)
+        unless error
           raise StripeError.new("Indeterminate error")
         end
 
       rescue JSON::ParserError, StripeError
         raise general_api_error(http_resp[:status], http_resp[:body])
       end
 
+      if error.is_a?(Hash)
+        error = specific_api_error(resp, error)
+      elsif error.is_a?(String)
+        error = oauth_error(resp, error)
+      else
+        error = StripeError.new(
+          "Indeterminate error",
+          http_status: resp.http_status, http_body: resp.http_body,
+          json_body: resp.data, http_headers: resp.http_headers
+        )
+      end
```

Comment:
Thanks for the fixes @ob-stripe!

Okay this is going to seem a like a major nit, but I have a little bit of FUD around making anything that has a string `error` field into an OAuth error just in case we add some new case in the future, or have missed a case that we already have (say an Nginx server somewhere returns something like this for the API).

What do you think of reversing this a little bit with something like:

```
# the error_description field is specific to OAuth errors
if resp.data[:error_description]
  error = oauth_error(resp, error)
elsif error.is_a?(Hash)
  error = specific_api_error(resp, error)
else
  ...
end
```

It kind of drives me nuts that we chose different formats for both types of errors in the first place, and that neither has a more distinguishing characteristic that we can more easily switch off of.